modules/cdc-ext/src/test/java/org/apache/ignite/cdc/CacheConflictOperationsTest.java

```diff
@@ -0,0 +1,267 @@
+    }
+
+
+    /**
+     * Tests that {@code IgniteInternalCache#*AllConflict} cache operations works with the conflict resolver
+     * when there is no update conflicts.
+     */
+    @Test
+    public void testUpdatesReorderFromOtherCluster() throws Exception {
+        String key = "UpdateClusterUpdateReorder";
+
+        putx(key(key, otherClusterId), otherClusterId, 2, true);
+
+        // Update with the equal or lower order should fail.
+        putx(key(key, otherClusterId), otherClusterId, 2, false);
+        putx(key(key, otherClusterId), otherClusterId, 1, false);
+
+        // Remove with the equal or lower order should fail.
+        removex(key(key, otherClusterId), otherClusterId, 2, false);
+        removex(key(key, otherClusterId), otherClusterId, 1, false);
+
+        // Remove with the higher order should succeed.
+        putx(key(key, otherClusterId), otherClusterId, 3, true);
+    }
+
+    /** Tests cache operations for entry replicated from another cluster. */
+    @Test
+    public void testUpdatesConflict() throws Exception {
+        String key = "UpdateThisClusterConflict0";
+
+        putx(key(key, otherClusterId), otherClusterId, 1, true);
+
+        // Local remove for other cluster entry should succeed.
+        remove(key(key, otherClusterId));
+
+        // Conflict replicated update succeed only if cluster has a greater priority than this cluster.
+        putx(key(key, otherClusterId), otherClusterId, 2, otherClusterId == FIRST_CLUSTER_ID);
+
+        key = "UpdateThisDCConflict1";
+
+        putx(key(key, otherClusterId), otherClusterId, 3, true);
+
+        // Local update for other cluster entry should succeed.
+        put(key(key, otherClusterId));
+
+        key = "UpdateThisDCConflict2";
+
+        put(key(key, otherClusterId));
+
+        // Conflict replicated remove succeed only if DC has a greater priority than this DC.
+        removex(key(key, otherClusterId), otherClusterId, 4, otherClusterId == FIRST_CLUSTER_ID);
+
+        key = "UpdateThisDCConflict3";
+
+        put(key(key, otherClusterId));
+
+        // Conflict replicated update succeed only if DC has a greater priority than this DC.
+        putx(key(key, otherClusterId), otherClusterId, 5, otherClusterId == FIRST_CLUSTER_ID || conflictResolveField() != null);
+    }
+
+    /** */
+    private void put(String key) {
+        Data newVal = Data.create();
+
+        cache.put(key, newVal);
+
+        assertEquals(newVal, cache.get(key));
+    }
+
+    /** */
+    private void putx(String k, byte otherClusterId, long order, boolean expectSuccess) throws IgniteCheckedException {
+        Data oldVal = cache.get(k);
+        Data newVal = Data.create();
+
+        KeyCacheObject key = new KeyCacheObjectImpl(k, null, cachex.context().affinity().partition(k));
+        CacheObject val = new CacheObjectImpl(client.binary().toBinary(newVal), null);
+
+        GridCacheVersion ver = new GridCacheVersion(1, order, 1, otherClusterId);
```

Comment:
Version 1 does not look correct here.
The old value will always win on version check?

Reply:
> The old value will always win on version check?

Check result depends on `order` value.